appsec/src/extension/commands_helpers.c

```diff
@@ -553,6 +559,41 @@ static void _set_appsec_span_data(mpack_node_t node)
     }
 }
 
+void dd_command_process_settings(mpack_node_t root)
+{
+    if (mpack_node_type(root) != mpack_type_map) {
+        return;
+    }
+
+    size_t count = mpack_node_map_count(root);
+
+    for (size_t i = 0; i < count; i++) {
+        mpack_node_t key = mpack_node_map_key_at(root, i);
+        mpack_node_t value = mpack_node_map_value_at(root, i);
+
+        if (mpack_node_type(key) != mpack_type_str) {
+            mlog(dd_log_warning,
+                "Failed to process setting: invalid type for key");
```

Comment:
Jus a nitpick, but since this function is user_collection related, I would add something to the log pointing to user_collection so it's easier to debug

Reply:
The `settings` field is meant for generic use, currently it's only used for user_collection but it could perfectly used for other purposes if we need to.

I can modify the log, but we need to change it once the `settings` field is used for something else

Reply:
Changed the log for now :+1: